environment/config.go

```diff
@@ -51,6 +55,41 @@ type Configuration struct {
 
 	// StaticConfiguration is used for running M3DB with a static config
 	Static *StaticConfiguration `yaml:"static"`
+
+	// Presence of a (etcd) server in this config denotes an embedded cluster
+	SeedNode *SeedNodeConfig `yaml:"seedNode"`
+
+	// NamespaceResolutionTimeout is the maximum time to wait to discover namespaces from KV
+	NamespaceResolutionTimeout time.Duration `yaml:"namespaceResolutionTimeout"`
+}
+
+// SeedNodeConfig defines fields for seed node
+type SeedNodeConfig struct {
+	RootDir                  string                 `yaml:"rootDir"`
+	InitialAdvertisePeerUrls []string               `yaml:"initialAdvertisePeerUrls"`
+	AdvertiseClientUrls      []string               `yaml:"advertiseClientUrls"`
+	ListenPeerUrls           []string               `yaml:"listenPeerUrls"`
+	ListenClientUrls         []string               `yaml:"listenClientUrls"`
+	InitialCluster           []SeedNode             `yaml:"initialCluster"`
+	Name                     string                 `yaml:"name"`
+	ClientTransportSecurity  SeedNodeSecurityConfig `json:"clientTransportSecurity"`
+	PeerTransportSecurity    SeedNodeSecurityConfig `json:"peerTransportSecurity"`
+}
+
+// SeedNode represents a seed node for the cluster
+type SeedNode struct {
+	HostID   string `yaml:"hostId"`
```

Comment:
nit: We usually keep the same field names except lower camel cased, so perhaps `hostID` here?  Also think we use `hostID` instead of `hostId` in other structs we use for YAML parsing.